tensorboard/components/tf_data_selector/tf-data-select-row.ts

```diff
@@ -0,0 +1,89 @@
+/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the 'License');
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an 'AS IS' BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+namespace tf_data_selector {
+
+Polymer({
+  is: 'tf-data-select-row',
+  properties: {
+    experiment: Object,
+
+    _runs: Array,
+
+    selectedRuns: {
+      type: Array,
+      notify: true,
+      value: () => [],
+    },
+
+    _runSelectionState: {
+      type: Object,
+      observer: '_storeRunSelectionState',
+      value: () => tf_storage.getObject('runSelectionState') || {},
+    },
+
+    _runRegexInput: {
+      type: String,
+      value: tf_storage.getStringInitializer('regexInput', {
+            defaultValue: '',
+            polymerProperty: '_runRegexInput',
+          }),
+      observer: '_storeRunRegexInput',
+    },
+  },
+
+  get _expsStore() {
+    return tf_backend.experimentsStore;
+  },
+
+  get _runsStore() {
+    return tf_backend.runsStore;
+  },
+
+  attached() {
+    this._updateExpKey = this._expsStore.addListener(() => this._updateExps());
+    this._updateRunKey = this._runsStore.addListener(() => this._updateRuns());
+
+    this._updateExps();
+    this._updateRuns();
+  },
+
+  detached() {
+    this._expsStore.removeListenerByKey(this._updateExpKey);
+    this._runsStore.removeListenerByKey(this._updateRunKey);
+  },
+
+  _updateExps() {
+    this.set('_experiments', this._expsStore.getExperiments());
+  },
+
+  _updateRuns() {
+    this.set('_runs', this._runsStore.getRuns());
```

Comment:
Just checking, but at some point this code will need to fetch the runs just for this experiment, right?  Since when comparing across experiments the sets of run names may be quite different.

Reply:
Correct. I am making incremental changes to plug other parts in so this is going to change a lot very soon (in a next branch)